source/Halibut.Tests/Transport/Streams/TimeoutStreamWrapperFixture.cs

```diff
@@ -0,0 +1,369 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut.Tests.Support;
+using Halibut.Tests.Support.TestAttributes;
+using Halibut.Transport.Streams;
+using NUnit.Framework;
+
+namespace Halibut.Tests.Transport.Streams
+{
+    [TestTimeout(timeoutInSeconds: 20)]
+    public class TimeoutStreamWrapperFixture : BaseTest
+    {
+        [Test]
+        public async Task ReadShouldPassThrough()
+        {
+            string? readData = null;
+            
+            var (disposables, sut, performListenerWrite) = await BuildTcpClientAndTcpListener(CancellationToken);
+
+            using (disposables)
+            {
+                await performListenerWrite("Test");
+
+                var buffer = new byte[19];
+                var readBytes = sut.Read(buffer, 0, 19);
+                if (readBytes > 0)
```

Comment:
Should we remove the `if`?

This test will fail if it does not read anything, so I'm wondering if it's worth guarding for.

Reply:
Or if it is important, should it be in a while loop until it reads everything?

Reply:
Yep it's not needed any more 👍